### PR TITLE
WIP: Also test against a newer OSX image

### PR DESCRIPTION
**Pull request by willirath**
This will probably fail because of #332

### CODE REVIEW

**Comment by willirath** [comment]
This PR is mostly to test if the error reported in #332 can be tracked down to the version of xcode.

**Comment by willirath** [comment]
(I've cancelled all but the two OSX builds that only differ in the OSX image.)

**Comment by willirath** [comment]
JIT fails with newest XCODE: <https://travis-ci.org/OceanParcels/parcels/jobs/366004275#L519>

**Comment by willirath** [comment]
21eac8c has no effect as the linker is never called explicitly.

File: .travis.yml [hunk 1]
```diff
@@ -23,6 +27,7 @@ install:
     - conda env create --name parcels --file environment_py${PY_VERSION}_${TRAVIS_OS_NAME}.yml
     - source activate parcels
     - python setup.py install
+    - ln -s ${LD} ${HOME}/miniconda/bin/ld # workaround avoiding os linker
```
Comment (willirath) [inline .travis.yml]: Note that this should go to the env's bin directory ...

**Comment by willirath** [comment]
With the last commit, I think I got what I wanted: Override the call to $CC with a call that makes sure the Anaconda linker is used. But that does not seem to help with xcode 9.3.

File: .travis.yml [hunk 2]
```diff
@@ -23,6 +27,7 @@ install:
     - conda env create --name parcels --file environment_py${PY_VERSION}_${TRAVIS_OS_NAME}.yml
     - source activate parcels
     - python setup.py install
+    - export CC=gcc
```
Comment (erikvansebille) [inline .travis.yml]: Should we not do this only when in `osx-xcode9.3` mode?

File: parcels/compiler.py [hunk 3]
```diff
@@ -28,13 +28,14 @@ class Compiler(object):
...
-        overridden by exporting the environment variable ``LDSHARED``).
+        overridden by exporting the environment variables ``LDSHARED`` or
+        ``LD``).
     :arg cppargs: A list of arguments to the C compiler (optional).
     :arg ldargs: A list of arguments to the linker (optional)."""
 
     def __init__(self, cc, ld=None, cppargs=[], ldargs=[]):
         self._cc = environ.get('CC', cc)
-        self._ld = environ.get('LDSHARED', ld)
+        self._ld = environ.get('LDSHARED', environ.get('LD', ld))
```
Comment (erikvansebille) [inline parcels/compiler.py]: These changes to `compiler.py` are not necessary, right? Do we still want to merge them into master?

**Comment by willirath** [comment]
I think we should close and drop this PR and just include an xcode 9.3 setup that is allowed to fail in the Travis config. The changes to the compiler.py did not lead anywhere.

**Pull request closed without merging**